public/app/features/explore/TraceView/components/model/transform-trace-data.tsx

```diff
@@ -74,7 +74,7 @@ export function orderTags(tags: TraceKeyValuePair[], topPrefixes?: string[]) {
  * generally requires.
  */
 export default function transformTraceData(data: TraceResponse | undefined): Trace | null {
-  if (!data?.traceID) {
+  if (!data?.traceID || data?.spans.some((x) => !x.processID || !x.spanID)) {
```

Comment:
I think this is too late to check this. We already processed a dataFrame response into `TraceResponse` but that transform should fail and not create something that is not a `TraceResponse`. So this check should be lifted up in the stack to more proper place so that we don't get type mismatches from a transform